scripts/python/command.py

```diff
@@ -0,0 +1,104 @@
+CONNECT = 1
+SET = 2
+ADD = 3
+SEND = 4
+JOIN = 5
+INVITE = 6
+KEY = 7
+ADDR = 8
+STOP = 9
+
+class CommandException(Exception):
+    pass
+
+string_to_cmd = {
+	"connect": CONNECT,
+	"set": SET,
+	"add": ADD,
+	"send": SEND,
+	"msg": SEND,
+	"join": JOIN,
+	"invite": INVITE,
+	"key": KEY,
+	"pubkey": KEY,
+	"address": ADDR,
+	"addr": ADDR,
+	"stop": STOP,
+}
+
+# check functions still need proper error handling
+def chk_connect(pssName, command, params):
+	if pssName != None:
+		raise CommandException("connect command isn't bound to a pssName")
+	if len(params) > 3:
+		raise CommandException("too many parameters")
+	if len(params) < 1:
+		raise CommandException("too few parameters")
+	
+	#TODO add check regex 
+
+def chk_set(params):
+	pass
+
+def chk_add(params):
+	pass
+
+def chk_send(params):
+	pass
+
+def chk_join(params):
+	pass
+
+def chk_invite(params):
+	pass
+
+def chk_key(params):
+	pass
+
+def chk_addr(params):
+	pass
+
+def chk_stop(params):
+	pass
+
+cmd_to_checker = {
```

Comment:
Maybe something along these lines will make things nice and compact - and auto failsafe?

```
#!/usr/bin/python2


fpostfixes = ["one", "two"]

def foo_one(v):
	print v
	pass


for f in fpostfixes:
	try:
		globals()["foo_" + f]("bar")
	except:
		print "no such func"	
```

Reply:
After looking into it a bit:

- Exception for not found function is **not** okay (yes we can add it but I think it's useless) that is "split" function duty.
If the command exists but we have no check function it means we aren't checking for errors. That's bad, we are treating user's input.
If the command doesn't exist the code must raise before arriving to check stage.

- Function call automation.
In this case: you would need strings to automate, correct?
It can be done using the command string itself but... what happens when you have an alias? Do you make a function for the alias? do you transform the "secondary" alias to the "main" one...

One could actually use:
```
CONNECT = "connect"
SET = "set"
```
... don't hate me :smiley:

Reply:
another solution could be
```
string_to_cmd = {
	"connect": "connect",
	"addr": "addr",
	"addr": "address",
}
```
but we lose the constants

Reply:
I am not sure I understand the reasoning here. This construct could be used to validate whether _command_ actually exists. Command only exists if there is a function to handle it. If there is no function, exception is raised and handling the exception means "invalid command." This way you don't keep a separate table for commands.

The split of the input would then only be to check whether you have to shift regarding to whether you are in "pss name context" or not.

Reply:
How do I know without checking the buffer if I'm in a pss context?

Without atleast a conversion dictionary, how do I make converge multiple strings to the same check function? (say, addr and address)